### DISCUSSION TITLE
Best way to understand if CXSMILES chiral flag is set?

### DISCUSSION THREAD

**Discussion by someonetookmyname** [rdkit/rdkit#7236] [Q&A] [answered]
Hi,

CXSMILES uses the "r" flag to indicate the chiral flag is unset, with its absence indicating the chiral flag is set.

However, MolFromSmiles() followed by MolToMolBlock() seems to generate the same mol data regardless of the presence/absence of the "r" flag.  See below.

Is there a way for _MolFileChiralFlag to be set automatically and appropriately?  Or a preferred way to understand if the CXSMILES has the "r" flag?  Or up to me to parse the CXSMILES?

Thank you for any guidance.

Best, Andrew

sp = Chem.SmilesParserParams()
sp.allowCXSMILES = True
sp.debugParse = False
sp.parseName = False
sp.removeHs = False
sp.sanitize = True
sp.strictCXSMILES = True

smiles = 'Cl[C@H]1CC[C@@H](Br)CC1 |r|'
m = Chem.MolFromSmiles(smiles, sp)
print(Chem.MolToMolBlock(m))

```

     RDKit          2D

  8  8  0  0  0  0  0  0  0  0999 V2000
    3.0000    0.0000    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
    1.5000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7500   -1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.7500   -1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.5000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.0000    0.0000    0.0000 Br  0  0  0  0  0  0  0  0  0  0  0  0
   -0.7500    1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7500    1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  1
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  1
  5  7  1  0
  7  8  1  0
  8  2  1  0
M  END
```

smiles = 'Cl[C@H]1CC[C@@H](Br)CC1'
m = Chem.MolFromSmiles(smiles, sp)
print(Chem.MolToMolBlock(m))

```

     RDKit          2D

  8  8  0  0  0  0  0  0  0  0999 V2000
    3.0000    0.0000    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
    1.5000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7500   -1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.7500   -1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.5000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -3.0000    0.0000    0.0000 Br  0  0  0  0  0  0  0  0  0  0  0  0
   -0.7500    1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.7500    1.2990    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  1
  2  3  1  0
  3  4  1  0
  4  5  1  0
  5  6  1  1
  5  7  1  0
  7  8  1  0
  8  2  1  0
M  END
```

**Comment by someonetookmyname** [comment, marked as answer]
Answered my own question .. according to https://www.rdkit.org/docs/RDKit_Book.html#cxsmiles-cxsmarts-extensions there is no mention of the "r" flag yet being supported in rdkit.

**Reply by greglandrum** [reply]
Correct. We don't currently read/write that flag.
If it's actually useful, we certainly could do so.

**Reply by someonetookmyname** [reply]
Hi Greg,

I think it would be very useful to be able to read/write it.  I think use/interpretation of the flag will be more complicated, as I haven't found documentation covering how that flag (or the mol-format global chiral flag) interacts with structural features and/or enhanced stererochemistry labels (for example, "r" flag is unset, but all stereocenters have "or" groups .. I'm not sure what that would mean).

But nonetheless being able to read and set the flag would be very useful.

Thanks, Andrew

**Reply by jrsacher** [reply]
Chiming in to say I'd love this feature! Right now, I just manually set/unset the chiral flag or `&` centers based on the presence of `r` in the CXSMILES string (depending on what I need). The process can be a bit annoying.
Realistically, though, I think reading the `r` flag would have to be optional. I've seen way too many files with pure enantiomers that have `|r|` because someone went from V2000 without a chiral flag to CXSMILES without double checking anything.
